## Projeto de Análise de Dados — Acidentes de Trânsito

##### 1. Contextualização

Os acidentes de trânsito representam um problema relevante de segurança pública e podem gerar impactos sociais e econômicos. A análise de dados permite identificar padrões relacionados à ocorrência desses acidentes e contribuir para uma melhor compreensão do problema.

Neste projeto, serão analisados dados públicos da Polícia Rodoviária Federal (PRF) referentes aos acidentes registrados em rodovias federais brasileiras entre 2023 e 2025.

#### 2. Objetivo

Identificar padrões, tendências e características relacionadas aos acidentes de trânsito, considerando aspectos como localização, período, causas, tipos e classificação das ocorrências.

Inicialmente, será realizada uma análise dos dados em âmbito nacional. A partir dos resultados encontrados, serão selecionados estados que apresentem volume e qualidade de dados adequados para uma análise mais aprofundada.

#### 3. Fonte de dados

Os dados utilizados são provenientes da base de dados abertos da Polícia Rodoviária Federal (PRF), contendo registros de acidentes ocorridos em rodovias federais brasileiras.

Foram utilizados os arquivos referentes aos anos de 2023, 2024 e 2025, disponibilizados em formato CSV.

Os dados são estruturados, organizados em linhas e colunas, contendo informações sobre localização, data, horário, características das vias, condições do acidente, causas, vítimas e veículos envolvidos.

A coleta foi realizada por meio do download direto dos arquivos disponibilizados publicamente pela PRF, que posteriormente foram importados para o Google Colab utilizando a biblioteca Pandas.

#### 4. Tecnologias utilizadas

- Python
- Pandas
- Google Colab
- Google Drive
- Looker Studio

#### 5. Importação dos dados

Os arquivos CSV foram importados utilizando a biblioteca Pandas. Como os arquivos utilizam `;` como separador e a codificação `latin-1`, esses parâmetros foram especificados durante a leitura.

In [ ]:
import pandas as pd

In [ ]:
df_23 = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/projeto semantix/datatran2023.csv', encoding='latin-1', sep=';')
print(df_23.head())

       id data_inversa dia_semana   horario  uf   br     km  \
0  496519   2023-01-01    domingo  02:00:00  ES  101    114   
1  496543   2023-01-01    domingo  03:40:00  SP  116  113,1   
2  496590   2023-01-01    domingo  01:40:00  MT  163   1112   
3  496610   2023-01-01    domingo  10:40:00  PR  376  314,8   
4  496659   2023-01-01    domingo  14:55:00  MG  116  569,4   

           municipio                            causa_acidente  \
0          SOORETAMA            Ausência de reação do condutor   
1            TAUBATE             Entrada inopinada do pedestre   
2  GUARANTA DO NORTE  Reação tardia ou ineficiente do condutor   
3         ORTIGUEIRA                   Velocidade Incompatível   
4           MANHUACU         Acumulo de água sobre o pavimento   

               tipo_acidente  ... feridos_graves ilesos ignorados feridos  \
0  Saída de leito carroçável  ...              0      0         0       1   
1  Atropelamento de Pedestre  ...              0      0         4     

In [ ]:
df_24 = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/projeto semantix/datatran2024.csv', encoding='latin-1', sep=';')
print(df_24.head())

         id data_inversa     dia_semana   horario  uf   br   km  \
0  571789.0   2024-01-01  segunda-feira  03:56:00  ES  101   38   
1  571804.0   2024-01-01  segunda-feira  04:50:00  PI  343  185   
2  571806.0   2024-01-01  segunda-feira  04:30:00  BA  116  578   
3  571818.0   2024-01-01  segunda-feira  06:30:00  SE  101   18   
4  571838.0   2024-01-01  segunda-feira  05:00:00  MT  364  240   

            municipio                                     causa_acidente  \
0  CONCEICAO DA BARRA                             Ultrapassagem Indevida   
1            PIRIPIRI                        Manobra de mudança de faixa   
2             BREJOES                   Ingestão de álcool pelo condutor   
3    MALHADA DOS BOIS           Reação tardia ou ineficiente do condutor   
4        RONDONOPOLIS  Condutor deixou de manter distância do veículo...   

                    tipo_acidente  ... feridos_graves ilesos ignorados  \
0  Colisão lateral sentido oposto  ...              1      1      

In [ ]:
df_25 = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/projeto semantix/datatran2025.csv', encoding='latin-1', sep=';')
print(df_25.head())

       id data_inversa    dia_semana   horario  uf   br     km  \
0  652493   2025-01-01  quarta-feira  06:20:00  SP  116    225   
1  652519   2025-01-01  quarta-feira  07:50:00  CE  116  546,2   
2  652522   2025-01-01  quarta-feira  08:45:00  PR  369   88,2   
3  652544   2025-01-01  quarta-feira  11:00:00  PR  116     74   
4  652549   2025-01-01  quarta-feira  09:30:00  MG  251    471   

               municipio                            causa_acidente  \
0              GUARULHOS  Reação tardia ou ineficiente do condutor   
1              PENAFORTE                          Pista esburacada   
2      CORNELIO PROCOPIO  Reação tardia ou ineficiente do condutor   
3  CAMPINA GRANDE DO SUL  Reação tardia ou ineficiente do condutor   
4           FRANCISCO SA                   Velocidade Incompatível   

               tipo_acidente  ... feridos_graves ilesos ignorados feridos  \
0                 Tombamento  ...              0      0         1       1   
1            Colisão frontal


### 6. Conhecendo os dados

Nesta etapa será realizada uma análise inicial da estrutura das bases, verificando a quantidade de registros, o número de variáveis e os tipos de dados presentes em cada arquivo.

In [ ]:
# 6.1 Quantidade de registros e variáveis

print('df_23:',df_23.shape)
print('df_24:', df_24.shape)
print('df_25:', df_25.shape)

df_23: (67766, 30)
df_24: (73156, 30)
df_25: (72529, 30)


#### 6.2 Variáveis disponíveis

A identificação das variáveis permite compreender quais informações estão disponíveis para a análise e verificar se a estrutura das bases permaneceu consistente entre os diferentes anos.

In [ ]:
print('Colunas de 2023:')
print(df_23.columns.tolist())

print('\nColunas de 2024:')
print(df_24.columns.tolist())

print('\nColunas de 2025:')
print(df_25.columns.tolist())

Colunas de 2023:
['id', 'data_inversa', 'dia_semana', 'horario', 'uf', 'br', 'km', 'municipio', 'causa_acidente', 'tipo_acidente', 'classificacao_acidente', 'fase_dia', 'sentido_via', 'condicao_metereologica', 'tipo_pista', 'tracado_via', 'uso_solo', 'pessoas', 'mortos', 'feridos_leves', 'feridos_graves', 'ilesos', 'ignorados', 'feridos', 'veiculos', 'latitude', 'longitude', 'regional', 'delegacia', 'uop']

Colunas de 2024:
['id', 'data_inversa', 'dia_semana', 'horario', 'uf', 'br', 'km', 'municipio', 'causa_acidente', 'tipo_acidente', 'classificacao_acidente', 'fase_dia', 'sentido_via', 'condicao_metereologica', 'tipo_pista', 'tracado_via', 'uso_solo', 'pessoas', 'mortos', 'feridos_leves', 'feridos_graves', 'ilesos', 'ignorados', 'feridos', 'veiculos', 'latitude', 'longitude', 'regional', 'delegacia', 'uop']

Colunas de 2025:
['id', 'data_inversa', 'dia_semana', 'horario', 'uf', 'br', 'km', 'municipio', 'causa_acidente', 'tipo_acidente', 'classificacao_acidente', 'fase_dia', 'sentido_

#### 6.3 Tipos de dados

Nesta etapa serão verificados os tipos de dados atribuídos às variáveis pelo Pandas. Essa verificação é importante para identificar possíveis inconsistências, como datas ou horários armazenados como texto, que poderão exigir tratamento posteriormente.

In [ ]:
df_23.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67766 entries, 0 to 67765
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   id                      67766 non-null  int64 
 1   data_inversa            67766 non-null  object
 2   dia_semana              67766 non-null  object
 3   horario                 67766 non-null  object
 4   uf                      67766 non-null  object
 5   br                      67766 non-null  int64 
 6   km                      67766 non-null  object
 7   municipio               67766 non-null  object
 8   causa_acidente          67766 non-null  object
 9   tipo_acidente           67766 non-null  object
 10  classificacao_acidente  67765 non-null  object
 11  fase_dia                67766 non-null  object
 12  sentido_via             67766 non-null  object
 13  condicao_metereologica  67766 non-null  object
 14  tipo_pista              67766 non-null  object
 15  tr

In [ ]:
df_24.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73156 entries, 0 to 73155
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      73156 non-null  float64
 1   data_inversa            73156 non-null  object 
 2   dia_semana              73156 non-null  object 
 3   horario                 73156 non-null  object 
 4   uf                      73156 non-null  object 
 5   br                      73156 non-null  int64  
 6   km                      73156 non-null  object 
 7   municipio               73156 non-null  object 
 8   causa_acidente          73156 non-null  object 
 9   tipo_acidente           73156 non-null  object 
 10  classificacao_acidente  73155 non-null  object 
 11  fase_dia                73156 non-null  object 
 12  sentido_via             73156 non-null  object 
 13  condicao_metereologica  73156 non-null  object 
 14  tipo_pista              73156 non-null

In [ ]:

df_25.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72529 entries, 0 to 72528
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   id                      72529 non-null  int64 
 1   data_inversa            72529 non-null  object
 2   dia_semana              72529 non-null  object
 3   horario                 72529 non-null  object
 4   uf                      72529 non-null  object
 5   br                      72529 non-null  int64 
 6   km                      72529 non-null  object
 7   municipio               72529 non-null  object
 8   causa_acidente          72529 non-null  object
 9   tipo_acidente           72529 non-null  object
 10  classificacao_acidente  72528 non-null  object
 11  fase_dia                72529 non-null  object
 12  sentido_via             72529 non-null  object
 13  condicao_metereologica  72529 non-null  object
 14  tipo_pista              72529 non-null  object
 15  tr

#### 6.4 Considerações iniciais sobre a estrutura dos dados

As bases analisadas apresentam a mesma estrutura nos três anos, contendo 30 variáveis cada. O conjunto de dados possui 67.766 registros em 2023, 73.156 em 2024 e 72.529 em 2025, totalizando 213.451 registros.

A padronização das colunas entre os anos permite a posterior consolidação das bases para realização de análises comparativas e temporais.

Durante a análise inicial, foram identificadas algumas diferenças nos tipos de dados entre os arquivos. A variável `id`, por exemplo, está armazenada como inteiro nos arquivos de 2023 e 2025, mas como número decimal em 2024. Também foram observadas diferenças nos tipos das variáveis `latitude` e `longitude`, que estão armazenadas como texto em alguns anos e como números em outro.

Além disso, foram identificados poucos valores ausentes em algumas variáveis, principalmente `classificacao_acidente`, `regional`, `delegacia` e `uop`. A quantidade e a distribuição desses valores serão investigadas na etapa de qualidade dos dados antes da definição das estratégias de tratamento.

#### 7.0 Qualidade dos dados

Nesta etapa serão avaliadas possíveis inconsistências presentes nas bases, incluindo valores ausentes, registros duplicados e diferenças de formato ou tipo das variáveis.

A identificação desses problemas é necessária para garantir maior consistência nas análises realizadas posteriormente.

#### 8.0 Valores ausentes

Inicialmente, será verificada a quantidade de valores ausentes em cada variável dos três conjuntos de dados.

In [ ]:
print('Valores ausentes - 2023')
print(df_23.isnull().sum())

Valores ausentes - 2023
id                         0
data_inversa               0
dia_semana                 0
horario                    0
uf                         0
br                         0
km                         0
municipio                  0
causa_acidente             0
tipo_acidente              0
classificacao_acidente     1
fase_dia                   0
sentido_via                0
condicao_metereologica     0
tipo_pista                 0
tracado_via                0
uso_solo                   0
pessoas                    0
mortos                     0
feridos_leves              0
feridos_graves             0
ilesos                     0
ignorados                  0
feridos                    0
veiculos                   0
latitude                   0
longitude                  0
regional                  17
delegacia                 32
uop                       46
dtype: int64


In [ ]:
print('\nValores ausentes - 2024')
print(df_24.isnull().sum())


Valores ausentes - 2024
id                          0
data_inversa                0
dia_semana                  0
horario                     0
uf                          0
br                          0
km                          0
municipio                   0
causa_acidente              0
tipo_acidente               0
classificacao_acidente      1
fase_dia                    0
sentido_via                 0
condicao_metereologica      0
tipo_pista                  0
tracado_via                 0
uso_solo                    0
pessoas                     0
mortos                      0
feridos_leves               0
feridos_graves              0
ilesos                      0
ignorados                   0
feridos                     0
veiculos                    0
latitude                    0
longitude                   0
regional                    3
delegacia                  36
uop                       112
dtype: int64


In [ ]:
print('\nValores ausentes - 2025')
print(df_25.isnull().sum())


Valores ausentes - 2025
id                         0
data_inversa               0
dia_semana                 0
horario                    0
uf                         0
br                         0
km                         0
municipio                  0
causa_acidente             0
tipo_acidente              0
classificacao_acidente     1
fase_dia                   0
sentido_via                0
condicao_metereologica     0
tipo_pista                 0
tracado_via                0
uso_solo                   0
pessoas                    0
mortos                     0
feridos_leves              0
feridos_graves             0
ilesos                     0
ignorados                  0
feridos                    0
veiculos                   0
latitude                   0
longitude                  0
regional                   2
delegacia                 22
uop                       38
dtype: int64


#### 8.1 Análise dos valores ausentes

A análise dos valores ausentes mostrou que as bases apresentam elevado nível de completude. As variáveis relacionadas às características dos acidentes, como `uf`, `causa_acidente`, `tipo_acidente`, `classificacao_acidente`, `pessoas`, `mortos` e `feridos`, apresentam praticamente todos os registros preenchidos.

Os valores ausentes estão concentrados principalmente nas variáveis `regional`, `delegacia` e `uop`, que possuem caráter administrativo e operacional. A maior quantidade identificada foi de 112 registros ausentes na variável `uop` em 2024, correspondendo a uma pequena parcela do total de registros daquele ano.

Diante da baixa quantidade de valores ausentes, não será realizada a exclusão dos registros nesta etapa. Os dados serão mantidos na base e, quando necessário, os valores ausentes serão tratados de acordo com a variável e com a finalidade de cada análise.

#### 8.1.1 Distribuição dos registros com valores ausentes

Será verificado se os valores ausentes estão concentrados nos mesmos registros ou distribuídos entre diferentes ocorrências.

In [ ]:
print('Registros com pelo menos um valor ausente:')
print('2023:', df_23.isnull().any(axis=1).sum())
print('2024:',df_24.isnull().any(axis=1).sum())
print('2025:', df_25.isnull().any(axis=1).sum())

print('Percentual de registros com pelo menos um valor ausente:')
print('2023:', round(df_23.isnull().any(axis=1).mean() * 100,3),'%')
print('2024:', round(df_24.isnull().any(axis=1).mean() * 100,3), '%')
print('2025:', round(df_25.isnull().any(axis=1).mean() * 100,3), '%')

Registros com pelo menos um valor ausente:
2023: 47
2024: 113
2025: 39
Percentual de registros com pelo menos um valor ausente:
2023: 0.069 %
2024: 0.154 %
2025: 0.054 %


Os registros que apresentam pelo menos um valor ausente representam uma parcela muito pequena das respectivas bases: 0,069% em 2023, 0,154% em 2024 e 0,054% em 2025.

Dessa forma, os dados apresentam elevado nível de completude, não sendo necessária a exclusão dos registros por ausência de informações. Os registros serão preservados e os valores ausentes poderão ser tratados pontualmente caso alguma análise específica utilize as variáveis afetadas.

#### 8.2 Registros duplicados

Nesta etapa será verificada a existência de registros duplicados nas bases. Inicialmente, serão identificadas linhas completamente iguais, ou seja, registros em que todas as variáveis apresentam exatamente os mesmos valores.

In [ ]:
print('Duplicidades - 2023')
print(df_23.duplicated().sum())

print('\nDuplicidade - 2024')
print(df_24.duplicated().sum())

print('\nDuplicidade - 2025')
print(df_25.duplicated().sum())

print('\nId duplicado 2023')
print(df_23['id'].duplicated().sum())

print('\nId duplicado 2024')
print(df_24['id'].duplicated().sum())

print('\nId duplicado 2025')
print(df_25['id'].duplicated().sum())


Duplicidades - 2023
0

Duplicidade - 2024
0

Duplicidade - 2025
0

Id duplicado 2023
0

Id duplicado 2024
0

Id duplicado 2025
0


#### 8.2.1 Resultado da análise de duplicidades

A análise não identificou registros completamente duplicados em nenhum dos três conjuntos de dados. Também não foram encontrados identificadores (`id`) duplicados dentro das bases de 2023, 2024 e 2025.

Dessa forma, não foi identificada a necessidade de remoção de registros por duplicidade nesta etapa.

#### 8.3 Verificação das variáveis numéricas

Durante a análise da estrutura dos dados foram identificadas diferenças nos tipos atribuídos a algumas variáveis entre os anos. Nesta etapa serão investigados os valores presentes nessas colunas para determinar se será necessário realizar uma padronização antes da análise.

In [ ]:
print('Tipo de ID')

print('2023:', df_23.dtypes['id'])
print('2024:', df_24.dtypes['id'])
print('2025:', df_25.dtypes['id'])

print('\nValores de ID com casas decimais em 2024:')
print(df_24['id'].head(10))

# latitude e longitude

print('\nTipo de Latitude')
print('2023:', df_23['latitude'].head(5).tolist())
print('2024:', df_24['latitude'].head(5).tolist())
print('2025:', df_25['latitude'].head(5).tolist())

print('\nTipo de Longitude')
print('2023:', df_23['longitude'].head(5).tolist())
print('2024:', df_24['longitude'].head(5).tolist())
print('2025:', df_25['longitude'].head(5).tolist())

print('\nExemplo de KM:')
print('2023:', df_23['km'].head(10).tolist())
print('2024:', df_24['km'].head(10).tolist())
print('2025:', df_25['km'].head(10).tolist())

Tipo de ID
2023: int64
2024: float64
2025: int64

Valores de ID com casas decimais em 2024:
0    571789.0
1    571804.0
2    571806.0
3    571818.0
4    571838.0
5    571855.0
6    571910.0
7    571922.0
8    571949.0
9    571957.0
Name: id, dtype: float64

Tipo de Latitude
2023: ['-19,09484877', '-23,0445658', '-9,70020602', '-23,985512', '-20,10007457']
2024: [-18.48261, -4.29603281, -13.07158302, -10.35601949, -16.17914141]
2025: ['-23,48586772', '-7,812288', '-23,182565', '-25,36517687', '-16,46801304']

Tipo de Longitude
2023: ['-40,05095848', '-45,58259814', '-54,87588757', '-51,083555', '-42,17884091']
2024: [-39.92379, -41.76732659, -39.9611107, -36.90552235, -54.78905337]
2025: ['-46,54075317', '-39,08333306', '-50,637228', '-49,04223028', '-43,43121303']

Exemplo de KM:
2023: ['114', '113,1', '1112', '314,8', '569,4', '569,8', '152', '897,3', '105,5', '366']
2024: ['38', '185', '578', '18', '240', '447', '852,9', '459,7', '550', '594,4']
2025: ['225', '546,2', '88,2', '74', '

In [ ]:
# Segunda verificação antes das mudanças

print('--- ID 2024 ---')
print('Quantidade ID\'s não inteiros')
print((df_24['id'] % 1 != 0). sum())

print('\n --- LATITUDE ---')
print('Valores Nulos:')
print(
    df_23['longitude'].isnull().sum(),
    df_24['longitude'].isnull().sum(),
    df_25['longitude'].isnull().sum()
)

print('\n --- KM ---')
print('Valores Nulos:')
print(
    df_23['km'].isnull().sum(),
    df_24['km'].isnull().sum(),
    df_25['km'].isnull().sum()
)

# Verificação se existem valores estranhos como 'km50' 'não informado'

print('\nValores únicos não numéricos em KM:')

for ano, df in [('2023', df_23), ('2024', df_24), ('2025', df_25)]:
    km_teste = pd.to_numeric(
        df['km'].astype(str).str.replace(',', '.', regex=False),
        errors='coerce'
    )

    print(f'\n{ano}:')
    print(df.loc[km_teste.isna(), 'km'].value_counts())

--- ID 2024 ---
Quantidade ID's não inteiros
0

 --- LATITUDE ---
Valores Nulos:
0 0 0

 --- KM ---
Valores Nulos:
0 0 0

Valores únicos não numéricos em KM:

2023:
Series([], Name: count, dtype: int64)

2024:
Series([], Name: count, dtype: int64)

2025:
Series([], Name: count, dtype: int64)


#### 8.3.1 Resultado da Verificação das variáveis numéricas

Foram analisadas as variáveis id, latitude, longitude e km, considerando seus tipos de dados, valores nulos e formatos de representação.

A variável id apresentou tipo float64 apenas em 2024, enquanto nos demais anos foi carregada como int64. A verificação mostrou que nenhum identificador possui parte decimal, indicando uma diferença de representação dos dados, e não uma característica dos valores.

As variáveis latitude e longitude não apresentaram valores nulos. Entretanto, nos arquivos de 2023 e 2025 os valores utilizam vírgula como separador decimal e foram carregados como texto, enquanto em 2024 já estão em formato numérico.

A variável km também não apresentou valores nulos. Apesar de estar armazenada como texto nos três anos, os valores encontrados possuem representação numérica, utilizando vírgula como separador decimal em parte dos registros.

Dessa forma, será necessário padronizar os formatos dessas variáveis antes da consolidação dos dados, garantindo que os três anos utilizem os mesmos tipos de dados.

#### 8.3.1 — Padronização do id

In [ ]:
df_23['id'] = df_23['id'].astype('int64')
df_24['id'] = df_24['id'].astype('int64')
df_25['id'] = df_25['id'].astype('int64')

print('Tipo de Dados')
print('2023:', df_23['id'].dtype)
print('2024:', df_24['id'].dtype)
print('2025:', df_25['id'].dtype)

Tipo de Dados
2023: int64
2024: int64
2025: int64


#### 8.3.3 — Padronização de latitude e longitude

In [ ]:
# Padronização da latitude

df_23['latitude'] = pd.to_numeric(
    df_23['latitude'].str.replace(',', '.', regex=False)
)

df_24['latitude'] = pd.to_numeric(df_24['latitude'])

df_25['latitude'] = pd.to_numeric(
    df_25['latitude'].str.replace(',', '.', regex=False)
)

# Padronização da longitude

df_23['longitude'] = pd.to_numeric(
    df_23['longitude'].str.replace(',', '.', regex=False)
)

df_24['longitude'] = pd.to_numeric(df_24['longitude'])

df_25['longitude'] = pd.to_numeric(
    df_25['longitude'].str.replace(',', '.', regex=False)
)

print('\nTipos após tratamento:')
print('--- Latitude ---')
print('2023:', df_23['latitude'].dtype)
print('2024:', df_24['latitude'].dtype)
print('2025:', df_25['latitude'].dtype)
print('\n--- Longitude ---')
print('2023:', df_23['longitude'].dtype)
print('2024:', df_24['longitude'].dtype)
print('2025:', df_25['longitude'].dtype)


Tipos após tratamento:
--- Latitude ---
2023: float64
2024: float64
2025: float64

--- Longitude ---
2023: float64
2024: float64
2025: float64


#### 8.3.4 Padronização da variável `km`

A variável `km` representa o quilômetro da rodovia onde ocorreu o acidente. Durante a análise inicial, ela foi identificada como uma variável do tipo texto (`object`) nos três conjuntos de dados.

Antes da conversão para um formato numérico, serão analisados alguns valores da variável para identificar possíveis diferenças de representação, como o uso de vírgula ou ponto como separador decimal.

In [ ]:
print('Exemplos da variável KM:')

print('\n--- 2023 ---')
print(df_23['km'].head(10).tolist())

print('\n--- 2024 ---')
print(df_24['km'].head(10).tolist())

print('\n--- 2025 ---')
print(df_25['km'].head(10).tolist())

Exemplos da variável KM:

--- 2023 ---
['114', '113,1', '1112', '314,8', '569,4', '569,8', '152', '897,3', '105,5', '366']

--- 2024 ---
['38', '185', '578', '18', '240', '447', '852,9', '459,7', '550', '594,4']

--- 2025 ---
['225', '546,2', '88,2', '74', '471', '669', '376', '207,4', '708,5', '7,4']


#### 8.3.4.1 Análise do formato da variável `km`

Os valores da variável `km` estão armazenados como texto nos três conjuntos de dados. A análise dos registros mostrou que os valores podem apresentar casas decimais utilizando vírgula como separador, enquanto os valores inteiros não apresentam separador decimal.

Para permitir operações numéricas e manter a mesma representação entre os anos, a variável será convertida para um formato numérico, substituindo a vírgula decimal por ponto antes da conversão.

In [ ]:
# Padronização da variável km

df_23['km'] = pd.to_numeric(
    df_23['km'].str.replace(',', '.', regex=False)
)

df_24['km'] = pd.to_numeric(
    df_24['km'].str.replace(',', '.', regex=False)
)

df_25['km'] = pd.to_numeric(
    df_25['km'].str.replace(',', '.', regex=False)
)

print('Tipos após tratamento:')
print('2023:', df_23['km'].dtype)
print('2024:', df_24['km'].dtype)
print('2025:', df_25['km'].dtype)


Tipos após tratamento:
2023: float64
2024: float64
2025: float64


In [ ]:
# Valores nulos
print('Valores ausentes em km após tratamento')

print('2023:', df_23['km'].isnull().sum())
print('2024:', df_24['km'].isnull().sum())
print('2025:', df_25['km'].isnull().sum())

# Km após tratamento do tipo e dados nulos
print('Exemplo após tratamento')

print('\n2023:')
print(df_23['km'].head(10).tolist())

print('\n2024:')
print(df_24['km'].head(10).tolist())

print('\n2025:')
print(df_25['km'].head(10).tolist())

Valores ausentes em km após tratamento
2023: 0
2024: 0
2025: 0
Exemplo após tratamento

2023:
[114.0, 113.1, 1112.0, 314.8, 569.4, 569.8, 152.0, 897.3, 105.5, 366.0]

2024:
[38.0, 185.0, 578.0, 18.0, 240.0, 447.0, 852.9, 459.7, 550.0, 594.4]

2025:
[225.0, 546.2, 88.2, 74.0, 471.0, 669.0, 376.0, 207.4, 708.5, 7.4]


#### 8.3.4.2 Validação da padronização de `km`

Após a substituição da vírgula pelo ponto decimal e a conversão da variável `km` para o formato numérico, as três bases passaram a apresentar a variável como `float64`.

A validação não identificou novos valores ausentes após a conversão, mantendo-se zero valores ausentes em `km` nos três anos.

Dessa forma, a variável está padronizada e preparada para operações numéricas e análises posteriores.

#### 8.3.5 Padronização das variáveis de data e horário

As variáveis `data_inversa` e `horario` foram inicialmente identificadas como texto (`object`) nas três bases.

Como essas variáveis serão utilizadas posteriormente para análises temporais, será realizada a conversão da data para o formato `datetime`, permitindo a extração e análise de informações como ano, mês e período da ocorrência.

In [ ]:
# Visualizando data
print('Exemplo de Data:')

print('\n2023:')
print(df_23['data_inversa'].head().tolist())

print('\n2024:')
print(df_24['data_inversa'].head().tolist())

print('\n2025:')
print(df_25['data_inversa'].head().tolist())

# Visualizando horário

print('\nExemplo de horário:')

print('\n2023:')
print(df_23['horario'].head().tolist())

print('\n2024:')
print(df_24['horario'].head().tolist())

print('\n2025:')
print(df_25['horario'].head().tolist())

Exemplo de Data:

2023:
['2023-01-01', '2023-01-01', '2023-01-01', '2023-01-01', '2023-01-01']

2024:
['2024-01-01', '2024-01-01', '2024-01-01', '2024-01-01', '2024-01-01']

2025:
['2025-01-01', '2025-01-01', '2025-01-01', '2025-01-01', '2025-01-01']

Exemplo de horário:

2023:
['02:00:00', '03:40:00', '01:40:00', '10:40:00', '14:55:00']

2024:
['03:56:00', '04:50:00', '04:30:00', '06:30:00', '05:00:00']

2025:
['06:20:00', '07:50:00', '08:45:00', '11:00:00', '09:30:00']


#### 8.3.5.1 Padronização das variáveis de data e horário

As variáveis `data_inversa` e `horario` foram inicialmente identificadas como texto (`object`). A análise dos valores mostrou que as datas seguem o padrão `YYYY-MM-DD` e os horários seguem o padrão `HH:MM:SS` nos três anos.

Como essas variáveis serão utilizadas posteriormente em análises temporais, a variável `data_inversa` será convertida para o formato `datetime`. O horário será mantido em formato de texto neste momento, pois sua representação já está padronizada e poderá ser utilizada posteriormente para criação de agrupamentos por período do dia.

In [ ]:
# Padronização da data

df_23['data_inversa'] = pd.to_datetime(df_23['data_inversa'])
df_24['data_inversa'] = pd.to_datetime(df_24['data_inversa'])
df_25['data_inversa'] = pd.to_datetime(df_25['data_inversa'])

print('Tipos após tratamento:')
print('2023:', df_23['data_inversa'].dtype)
print('2024:', df_24['data_inversa'].dtype)
print('2025:', df_25['data_inversa'].dtype)

Tipos após tratamento:
2023: datetime64[ns]
2024: datetime64[ns]
2025: datetime64[ns]


In [ ]:
# Validar nulos

print('\nValores ausentes após conversão:')
print('2023:', df_23['data_inversa'].isnull().sum())
print('2024:', df_24['data_inversa'].isnull().sum())
print('2025:', df_25['data_inversa'].isnull().sum())


Valores ausentes após conversão:
2023: 0
2024: 0
2025: 0


In [ ]:
# Validando a coluna horario
print('Quantidade de horários diferentes por ano:')
print('2023:', df_23['horario'].nunique())
print('2024:', df_24['horario'].nunique())
print('2025:', df_25['horario'].nunique())

print('\nHorário mínimo e máximo:')

print('2023:')
print(df_23['horario'].min(), 'até', df_23['horario'].max())

print('\n2024:')
print(df_24['horario'].min(), 'até', df_24['horario'].max())

print('\n2025:')
print(df_25['horario'].min(), 'até', df_25['horario'].max())

Quantidade de horários diferentes por ano:
2023: 1406
2024: 1415
2025: 1412

Horário mínimo e máximo:
2023:
00:00:00 até 23:59:00

2024:
00:00:00 até 23:59:00

2025:
00:00:00 até 23:59:00


#### 8.3.6 — Verificação das variáveis numéricas

Após a padronização das variáveis `id`, `latitude`, `longitude`, `km`, `data_inversa` e `horario`, será realizada uma verificação das demais variáveis quantitativas presentes nas bases.

Nesta etapa, serão analisadas as variáveis relacionadas à quantidade de pessoas, vítimas e veículos envolvidos nos acidentes: `pessoas`, `mortos`, `feridos_leves`, `feridos_graves`, `ilesos`, `ignorados`, `feridos` e `veiculos`.

O objetivo é verificar se essas variáveis apresentam valores ausentes, valores que não possam ser interpretados como números ou valores negativos, que seriam incompatíveis com a natureza dessas informações.

Essa análise é importante para identificar possíveis inconsistências nos dados antes da consolidação dos três anos, garantindo que as variáveis quantitativas estejam adequadas para cálculos, agrupamentos e análises estatísticas posteriores.

In [ ]:
# 8.3.6 — Verificação das variáveis numéricas

colunas_numericas = [
    'pessoas',
    'mortos',
    'feridos_leves',
    'feridos_graves',
    'ilesos',
    'ignorados',
    'feridos',
    'veiculos'
]

for ano, df in [('2023', df_23), ('2024', df_24), ('2025', df_25)]:
    print(f'\n--- {ano} ---')

    for coluna in colunas_numericas:
        valores = pd.to_numeric(df[coluna], errors='coerce')

        print(
            f'{coluna}: '
            f'nulos={df[coluna].isnull().sum()} | '
            f'não numéricos={valores.isnull().sum() - df[coluna].isnull().sum()} | '
            f'negativos={(valores < 0).sum()}'
        )


--- 2023 ---
pessoas: nulos=0 | não numéricos=0 | negativos=0
mortos: nulos=0 | não numéricos=0 | negativos=0
feridos_leves: nulos=0 | não numéricos=0 | negativos=0
feridos_graves: nulos=0 | não numéricos=0 | negativos=0
ilesos: nulos=0 | não numéricos=0 | negativos=0
ignorados: nulos=0 | não numéricos=0 | negativos=0
feridos: nulos=0 | não numéricos=0 | negativos=0
veiculos: nulos=0 | não numéricos=0 | negativos=0

--- 2024 ---
pessoas: nulos=0 | não numéricos=0 | negativos=0
mortos: nulos=0 | não numéricos=0 | negativos=0
feridos_leves: nulos=0 | não numéricos=0 | negativos=0
feridos_graves: nulos=0 | não numéricos=0 | negativos=0
ilesos: nulos=0 | não numéricos=0 | negativos=0
ignorados: nulos=0 | não numéricos=0 | negativos=0
feridos: nulos=0 | não numéricos=0 | negativos=0
veiculos: nulos=0 | não numéricos=0 | negativos=0

--- 2025 ---
pessoas: nulos=0 | não numéricos=0 | negativos=0
mortos: nulos=0 | não numéricos=0 | negativos=0
feridos_leves: nulos=0 | não numéricos=0 | negati

#### 8.3.6.1 — Resultado da verificação das variáveis numéricas

A verificação das variáveis quantitativas não identificou valores ausentes, registros não numéricos ou valores negativos em nenhuma das três bases.

As variáveis `pessoas`, `mortos`, `feridos_leves`, `feridos_graves`, `ilesos`, `ignorados`, `feridos` e `veiculos` apresentaram **zero ocorrências** para os três critérios analisados em 2023, 2024 e 2025.

Dessa forma, não foram identificadas inconsistências relacionadas aos valores dessas variáveis, não sendo necessário realizar tratamentos adicionais nesta etapa. Os dados estão adequados para as análises quantitativas posteriores.


#### 8.3.7 — Verificação da consistência entre variáveis

Após verificar individualmente as variáveis quantitativas, será analisada a consistência entre os valores registrados em diferentes colunas que representam informações relacionadas ao mesmo acidente.

Nesta etapa, será verificado se a quantidade total de `feridos` corresponde à soma de `feridos_leves` e `feridos_graves`. Também será analisado se a quantidade de `pessoas` corresponde à soma de `mortos`, `feridos_leves`, `feridos_graves`, `ilesos` e `ignorados`.

Essa verificação permite identificar possíveis inconsistências lógicas nos registros, mesmo quando os valores de cada variável, analisados isoladamente, estão preenchidos corretamente.

A análise será realizada separadamente para 2023, 2024 e 2025, permitindo avaliar se essas relações permanecem consistentes nos três conjuntos de dados antes da consolidação das bases.

In [ ]:
# 8.3.7 - Verificação de consisitências entre variáveis

for ano, df in [('2023', df_23),('2024', df_24),('2025', df_25)]:
  # Verificação: feridos = feridos leves + feridos graves
  inconsistencias_feridos = (
      df['feridos'] != (df['feridos_leves'] + df['feridos_graves'])
  ).sum()
  # Verificação: pessoas = mortos + feridos leves + feridos graves + ilesos + ignorados
  inconsistencias_pessoas = (
      df['pessoas'] != (
          df['mortos']
          + df['feridos_leves']
          + df['feridos_graves']
          + df['ilesos']
          + df['ignorados']
      )
  ).sum()

  print('\n--- Ano ---')
  print('Inconsistências em feridos:', inconsistencias_feridos)
  print('Inconsistências em pessoas:', inconsistencias_pessoas)


--- Ano ---
Inconsistências em feridos: 0
Inconsistências em pessoas: 3767

--- Ano ---
Inconsistências em feridos: 0
Inconsistências em pessoas: 3884

--- Ano ---
Inconsistências em feridos: 0
Inconsistências em pessoas: 3823


#### 8.3.7.1 — Resultado inicial da verificação de consistência

A verificação das relações entre as variáveis quantitativas mostrou que a variável `feridos` apresenta consistência nos três anos analisados. Em todos os registros, o total de feridos corresponde à soma de `feridos_leves` e `feridos_graves`.

Entretanto, foram identificadas diferenças entre a variável `pessoas` e a soma das variáveis `mortos`, `feridos_leves`, `feridos_graves`, `ilesos` e `ignorados`. Foram encontrados 3.767 registros com diferença em 2023, 3.884 em 2024 e 3.823 em 2025.

Como essas diferenças não indicam, por si só, que os registros estejam incorretos, será realizada uma investigação adicional para identificar a magnitude e o padrão dessas diferenças. Essa análise permitirá compreender se as ocorrências representam inconsistências nos dados ou alguma característica da forma como as informações foram registradas na base.

Nenhum dado será alterado nesta etapa. O tratamento, caso seja necessário, será definido somente após a compreensão da origem dessas diferenças.


In [ ]:
# Investigação das inconsistências na variável pessoas

for ano, df in [('2023', df_23), ('2024', df_24), ('2025', df_25)]:

    diferenca = (
        df['pessoas']
        - (
            df['mortos']
            + df['feridos_leves']
            + df['feridos_graves']
            + df['ilesos']
            + df['ignorados']
        )
    )

    print(f'\n--- {ano} ---')
    print('Quantidade de inconsistências:', (diferenca != 0).sum())
    print('Diferenças encontradas:')
    print(diferenca[diferenca != 0].value_counts().sort_index())


--- 2023 ---
Quantidade de inconsistências: 3767
Diferenças encontradas:
-48       1
-16       1
-14       1
-13       2
-12       2
-11      13
-10       8
-9        2
-8        7
-7       12
-6       16
-5       59
-4       71
-3      197
-2      996
-1     2379
Name: count, dtype: int64

--- 2024 ---
Quantidade de inconsistências: 3884
Diferenças encontradas:
-13       1
-12       1
-11      11
-10       9
-9        1
-8       15
-7       10
-6       11
-5       65
-4       86
-3      187
-2     1090
-1     2397
Name: count, dtype: int64

--- 2025 ---
Quantidade de inconsistências: 3823
Diferenças encontradas:
-80       1
-14       1
-13       1
-11      11
-10       5
-9        1
-8        9
-7       11
-6       14
-5       61
-4       85
-3      206
-2     1083
-1     2334
Name: count, dtype: int64


In [ ]:
# Investigação dos registros com inconsistência em pessoas

for ano, df in [('2023', df_23), ('2024', df_24), ('2025', df_25)]:

    soma_pessoas = (
        df['mortos']
        + df['feridos_leves']
        + df['feridos_graves']
        + df['ilesos']
        + df['ignorados']
    )

    diferenca = df['pessoas'] - soma_pessoas

    print(f'\n--- {ano} ---')

    print('\nExemplos com diferença de -1:')
    print(
        df.loc[
            diferenca == -1,
            ['id', 'pessoas', 'mortos', 'feridos_leves',
             'feridos_graves', 'ilesos', 'ignorados']
        ].head(10)
    )

    print('\nExemplos com maiores diferenças:')
    print(
        df.loc[
            diferenca <= -10,
            ['id', 'pessoas', 'mortos', 'feridos_leves',
             'feridos_graves', 'ilesos', 'ignorados']
        ].head(10)
    )


--- 2023 ---

Exemplos com diferença de -1:
        id  pessoas  mortos  feridos_leves  feridos_graves  ilesos  ignorados
2   496590        2       0              0               1       0          2
3   496610        2       0              0               0       1          2
23  496818        3       1              0               0       1          2
26  496834        4       1              1               1       0          2
30  496862        9       1              2               1       4          2
45  496950        4       0              1               0       2          2
47  496969        2       0              1               0       0          2
60  497025        5       0              1               1       2          2
61  497026        7       0              0               1       5          2
65  497038        3       1              0               0       1          2

Exemplos com maiores diferenças:
          id  pessoas  mortos  feridos_leves  feridos_graves  i

#### 8.3.7.2 — Resultado da análise de consistência entre variáveis

A análise confirmou que a variável `feridos` apresenta consistência nos três anos, sendo equivalente à soma de `feridos_leves` e `feridos_graves` em todos os registros analisados.

Inicialmente, também foi verificada a relação entre `pessoas` e a soma de `mortos`, `feridos_leves`, `feridos_graves`, `ilesos` e `ignorados`. Foram identificadas diferenças em parte dos registros. A investigação dos casos mostrou que essas diferenças estão relacionadas principalmente à variável `ignorados`, que apresenta valores que não devem ser interpretados como uma quantidade de pessoas a ser somada diretamente às demais categorias para reconstrução do total de `pessoas`.

Dessa forma, não serão realizadas alterações nessas variáveis. Os valores originais da fonte serão preservados, evitando a aplicação de uma regra de tratamento sem justificativa sobre o significado da variável `ignorados`.

A variável `feridos` apresentou consistência e está adequada para utilização nas análises posteriores. A variável `pessoas` também será mantida conforme os valores disponibilizados pela fonte.

#### 8.4 — Modelagem e preparação da base

Após a etapa de qualidade e padronização, foi realizada a preparação da estrutura de dados que será utilizada nas análises. O processo teve como objetivo transformar as três bases anuais, inicialmente separadas, em uma única base analítica padronizada.

As bases referentes aos anos de 2023, 2024 e 2025 possuem a mesma estrutura de variáveis e representam registros individuais de acidentes ocorridos em rodovias federais brasileiras. Dessa forma, foi adotado um modelo de dados consolidado, no qual cada linha representa uma ocorrência de acidente e as colunas representam suas respectivas características.

Antes da consolidação, foram padronizadas variáveis que apresentavam diferenças de tipo ou representação entre os anos, incluindo id, latitude, longitude, km e data_inversa. Também foram realizadas verificações de valores ausentes, duplicidades, valores numéricos e consistência entre variáveis relacionadas às vítimas.

Após essas etapas, as três bases foram empilhadas verticalmente, preservando as mesmas colunas e mantendo cada acidente como uma observação independente.

A consolidação foi realizada utilizando a biblioteca Pandas:


In [ ]:
df = pd.concat([df_23, df_24, df_25], ignore_index=True)

Como a variável data_inversa foi convertida para o formato datetime, foi criada a variável ano, permitindo identificar o período de cada ocorrência e realizar análises temporais:

In [ ]:
# 8.4.1 — Validação da base consolidada

print('Dimensão da base consolidada:')
print(df.shape)

print('\nQuantidade de registros por ano:')
print(df['data_inversa'].dt.year.value_counts().sort_index())

print('\nQuantidade de colunas:')
print(len(df.columns))

print('\nTipos de dados:')
print(df.dtypes)

Dimensão da base consolidada:
(213451, 30)

Quantidade de registros por ano:
data_inversa
2023    67766
2024    73156
2025    72529
Name: count, dtype: int64

Quantidade de colunas:
30

Tipos de dados:
id                                 int64
data_inversa              datetime64[ns]
dia_semana                        object
horario                           object
uf                                object
br                                 int64
km                               float64
municipio                         object
causa_acidente                    object
tipo_acidente                     object
classificacao_acidente            object
fase_dia                          object
sentido_via                       object
condicao_metereologica            object
tipo_pista                        object
tracado_via                       object
uso_solo                          object
pessoas                            int64
mortos                             int64
feridos_leves      

A base consolidada passou a reunir 213.451 registros, mantendo as 30 variáveis originais e incorporando a variável ano para facilitar os agrupamentos temporais.

Essa estrutura foi utilizada como base para a Análise Exploratória de Dados (EDA), permitindo comparar os acidentes entre os anos e analisar suas características por estado, causa e gravidade.

#### 8.5 — Análise Exploratória de Dados (EDA)

Com as bases de 2023, 2024 e 2025 consolidadas e as principais inconsistências verificadas e tratadas, inicia-se a etapa de Análise Exploratória de Dados (EDA).

O objetivo desta etapa é explorar os dados para identificar padrões, tendências e características relevantes dos acidentes registrados nas rodovias federais brasileiras durante o período analisado.

Inicialmente, será realizada uma visão geral dos acidentes, considerando a distribuição das ocorrências ao longo dos anos e algumas das principais medidas relacionadas às pessoas envolvidas e às vítimas.

A partir dos resultados obtidos, serão selecionadas as variáveis e os recortes mais relevantes para análises posteriores, buscando responder às questões definidas para o projeto e identificar possíveis insights que possam contribuir para a compreensão da problemática estudada.

In [ ]:
# 8.5.1 — Visão geral dos acidentes

print('Total de acidentes:', len(df))

print('\nAcidentes por ano:')
print(df['data_inversa'].dt.year.value_counts().sort_index())

print('\nTotal de mortos:', df['mortos'].sum())

print('Total de feridos:', df['feridos'].sum())

print('Total de pessoas envolvidas:', df['pessoas'].sum())

print('\nMédia de pessoas por acidente:', round(df['pessoas'].mean(), 2))

print('\nMédia de mortos por acidente:', round(df['mortos'].mean(), 2))

print('\nMédia de feridos por acidente:', round(df['feridos'].mean(), 2))

Total de acidentes: 213451

Acidentes por ano:
data_inversa
2023    67766
2024    73156
2025    72529
Name: count, dtype: int64

Total de mortos: 17830
Total de feridos: 246539
Total de pessoas envolvidas: 555559

Média de pessoas por acidente: 2.6

Média de mortos por acidente: 0.08

Média de feridos por acidente: 1.16


#### 8.5.1.1 — Resultado da visão geral dos acidentes

A base consolidada reúne **213.451 acidentes** registrados em rodovias federais brasileiras entre 2023 e 2025.

Em relação à distribuição anual, foram registrados 67.766 acidentes em 2023, 73.156 em 2024 e 72.529 em 2025. Observa-se, portanto, um aumento no número de registros de 2023 para 2024, seguido por uma pequena redução em 2025.

No período analisado, foram contabilizadas **17.830 mortes** e **246.539 feridos**, enquanto o total de pessoas envolvidas nos acidentes foi de **555.559**.

A média foi de **2,60 pessoas envolvidas por acidente** e **1,16 ferido por ocorrência**. A média de mortes foi de **0,08 por acidente**.

Esses resultados fornecem uma visão inicial da dimensão do problema e servirão como referência para as análises seguintes, nas quais serão investigados os períodos, locais, causas e características dos acidentes com o objetivo de identificar padrões e possíveis fatores associados à sua ocorrência e gravidade.

In [ ]:
# 8.5.2 — Acidentes fatais

acidentes_fatais = (df['mortos'] > 0).sum()
acidentes_nao_fatais = (df['mortos'] == 0).sum()

percentual_fatais = acidentes_fatais / len(df) * 100

print('Acidentes fatais:', acidentes_fatais)
print('Acidentes não fatais:', acidentes_nao_fatais)
print('Percentual de acidentes fatais:', round(percentual_fatais, 2), '%')

Acidentes fatais: 15290
Acidentes não fatais: 198161
Percentual de acidentes fatais: 7.16 %


#### 8.5.2 — Identificação dos acidentes fatais

Para complementar a visão geral dos acidentes, será analisada a quantidade de ocorrências que resultaram em pelo menos uma morte.

Um acidente será considerado fatal quando a variável `mortos` apresentar valor maior que zero. Essa classificação permite diferenciar a quantidade de acidentes fatais dos acidentes que não registraram mortes e calcular a proporção de ocorrências fatais no período analisado.

Essa métrica é relevante para avaliar a gravidade das ocorrências e, posteriormente, investigar quais características dos acidentes estão mais associadas à presença de vítimas fatais.

No período de 2023 a 2025, foram identificados **15.290 acidentes fatais**, correspondendo a **7,16% dos 213.451 acidentes analisados**. Os demais **198.161 acidentes**, equivalentes a **92,84%**, não registraram mortes.

A proporção de acidentes fatais será utilizada nas análises posteriores como uma das principais medidas para avaliar a gravidade das ocorrências.



In [ ]:
# 8.5.3 — Evolução anual dos acidentes

df['ano'] = df['data_inversa'].dt.year

evolucao_anual = df.groupby('ano').agg(
    acidentes=('id', 'count'),
    mortos=('mortos', 'sum'),
    feridos=('feridos', 'sum')
).reset_index()

acidentes_fatais_ano = (
    df[df['mortos'] > 0]
    .groupby('ano')
    .size()
    .reset_index(name='acidentes_fatais')
)

evolucao_anual = evolucao_anual.merge(
    acidentes_fatais_ano,
    on='ano',
    how='left'
)

evolucao_anual['percentual_fatais'] = (
    evolucao_anual['acidentes_fatais']
    / evolucao_anual['acidentes']
    * 100
)

print(evolucao_anual)

    ano  acidentes  mortos  feridos  acidentes_fatais  percentual_fatais
0  2023      67766    5627    78463              4858           7.168787
1  2024      73156    6160    84526              5222           7.138170
2  2025      72529    6043    83550              5210           7.183334


#### 8.5.3.1 — Resultado da evolução anual

A análise da distribuição anual permitiu comparar a quantidade de acidentes, mortes, feridos e acidentes fatais entre 2023 e 2025.

Em 2024 foi registrado o maior número de acidentes, com 73.156 ocorrências, além do maior número de mortes (6.160) e feridos (84.526). Em 2025, foram registrados 72.529 acidentes, 6.043 mortes e 83.550 feridos, enquanto 2023 apresentou os menores valores absolutos entre os três anos.

Apesar das variações nos números absolutos, a proporção de acidentes fatais apresentou comportamento bastante estável: 7,17% em 2023, 7,14% em 2024 e 7,18% em 2025.

Dessa forma, os dados indicam que, no período analisado, não houve uma mudança relevante na proporção de acidentes que resultaram em mortes. As diferenças observadas entre os anos estão principalmente relacionadas ao volume total de ocorrências, enquanto a proporção de acidentes fatais permaneceu próxima de 7,1%.

Esse resultado servirá como referência para as próximas análises, que buscarão identificar se determinadas características dos acidentes apresentam maior associação com ocorrências fatais.

In [ ]:
# 8.5.4 — Distribuição dos acidentes por estado

acidentes_uf = (
    df.groupby('uf').size().sort_values(ascending=False)
)

print('Quantidade de acidentes por UF')
print(acidentes_uf)

Quantidade de acidentes por UF
uf
MG    27873
SC    24366
PR    22316
RJ    18392
RS    15041
SP    14320
BA    12014
GO     9637
PE     9244
MT     7507
ES     7278
PB     5462
MS     5183
RN     4621
RO     4335
PI     4283
CE     4165
MA     3563
DF     3044
PA     2986
TO     2040
AL     1961
SE     1723
AC      792
AP      472
RR      421
AM      412
dtype: int64


#### 8.5.4.1 — Resultado da distribuição dos acidentes por estado

A análise da distribuição dos acidentes por unidade federativa mostrou diferenças relevantes no volume de ocorrências registradas entre os estados.

Minas Gerais apresentou o maior número de acidentes no período, com **27.873 registros**, seguido por Santa Catarina, com **24.366**, e Paraná, com **22.316**. Na sequência aparecem Rio de Janeiro, com **18.392**, Rio Grande do Sul, com **15.041**, e São Paulo, com **14.320** ocorrências.

Os resultados indicam uma concentração maior de registros em determinados estados. Entretanto, o número absoluto de acidentes não deve ser utilizado isoladamente como indicador de maior gravidade, pois diferenças no volume de ocorrências não significam necessariamente maior proporção de acidentes fatais.

Dessa forma, a distribuição por estado será utilizada como ponto de partida para análises posteriores que relacionem o volume de acidentes com indicadores de gravidade, como mortes e acidentes fatais.

In [ ]:
# 8.5.5 — Acidentes e gravidade por UF

acidentes_uf = (
    df.groupby('uf')
    .agg(
        acidentes=('id', 'count'),
        mortos=('mortos', 'sum')
    )
    .reset_index()
)

acidentes_fatais_uf = (
    df[df['mortos'] > 0]
    .groupby('uf')
    .size()
    .reset_index(name='acidentes_fatais')
)

acidentes_uf = acidentes_uf.merge(
    acidentes_fatais_uf,
    on='uf',
    how='left'
)

acidentes_uf['percentual_fatais'] = (
    acidentes_uf['acidentes_fatais']
    / acidentes_uf['acidentes']
    * 100
)

acidentes_uf = acidentes_uf.sort_values(
    'percentual_fatais',
    ascending=False
)

print(acidentes_uf)

    uf  acidentes  mortos  acidentes_fatais  percentual_fatais
9   MA       3563     812               698          19.590233
13  PA       2986     647               565          18.921634
21  RR        421      84                70          16.627078
2   AM        412      63                53          12.864078
1   AL       1961     257               232          11.830699
4   BA      12014    1766              1395          11.611453
26  TO       2040     296               236          11.568627
5   CE       4165     493               451          10.828331
16  PI       4283     488               439          10.249825
15  PE       9244     926               835           9.032886
11  MS       5183     516               427           8.238472
12  MT       7507     758               610           8.125749
0   AC        792      73                63           7.954545
24  SE       1723     148               134           7.777133
8   GO       9637     895               739           7

#### 8.5.5.1 — Resultado da análise de acidentes e gravidade por UF

A análise conjunta do volume de acidentes e dos indicadores de gravidade mostrou que a concentração de ocorrências não acompanha necessariamente a proporção de acidentes fatais.

O Maranhão apresentou a maior proporção de acidentes fatais entre os estados analisados, com **19,59%**, seguido pelo Pará, com **18,92%**. A Bahia também apresentou uma proporção elevada, com **11,61%** de acidentes fatais.

Por outro lado, estados com grande volume de ocorrências apresentaram proporções menores. Minas Gerais, que registrou o maior número de acidentes no período, apresentou **6,80%** de acidentes fatais. São Paulo apresentou **4,42%**, enquanto Santa Catarina apresentou **4,34%**.

Esse resultado evidencia a importância de analisar conjuntamente o volume de acidentes e sua gravidade. Um estado pode apresentar grande quantidade de ocorrências sem necessariamente apresentar a maior proporção de acidentes fatais.

Os estados com menor quantidade de registros devem ser interpretados com cautela, pois volumes reduzidos podem gerar maiores variações nas proporções calculadas. Essa consideração será levada em conta nas análises e na interpretação dos resultados.

#### 8.5.6 — Principais causas dos acidentes

Nesta etapa será analisada a distribuição dos acidentes de acordo com a variável `causa_acidente`.

O objetivo é identificar quais causas aparecem com maior frequência nos registros do período de 2023 a 2025. Essa análise permite compreender os principais fatores associados à ocorrência dos acidentes e selecionar as causas mais relevantes para investigações posteriores.

Inicialmente, será considerado o número absoluto de ocorrências por causa. Em uma etapa posterior, as principais causas poderão ser relacionadas à ocorrência de acidentes fatais, permitindo avaliar se determinadas causas apresentam maior participação entre as ocorrências com mortes.

Essa análise contribui para a identificação de padrões que poderão ser utilizados na elaboração dos insights e das visualizações do projeto.

In [ ]:
# 8.5.6 — Principais causas dos acidentes

causas_acidentes = (
    df['causa_acidente']
    .value_counts()
)

print('Quantidade de de acidentes por causa:')
print(causas_acidentes)

Quantidade de de acidentes por causa:
causa_acidente
Reação tardia ou ineficiente do condutor                     31574
Ausência de reação do condutor                               31459
Acessar a via sem observar a presença dos outros veículos    20380
Condutor deixou de manter distância do veículo da frente     13145
Velocidade Incompatível                                      12675
                                                             ...  
Redutor de velocidade em desacordo                              22
Faróis desregulados                                             22
Sinalização encoberta                                           14
Obstrução Via tentativa Assalto                                 10
Sistema de drenagem ineficiente                                  8
Name: count, Length: 75, dtype: int64


#### 8.5.6.1 — Resultado da análise das principais causas

A análise da variável `causa_acidente` identificou **75 categorias diferentes** de causas registradas nos acidentes analisados.

As causas com maior número de ocorrências foram **"Reação tardia ou ineficiente do condutor"**, com 31.574 registros, e **"Ausência de reação do condutor"**, com 31.459 registros. Em seguida, destaca-se **"Acessar a via sem observar a presença dos outros veículos"**, com 20.380 ocorrências.

Também apresentaram volumes relevantes as causas **"Condutor deixou de manter distância do veículo da frente"**, com 13.145 registros, e **"Velocidade Incompatível"**, com 12.675.

Os resultados mostram que algumas categorias concentram uma parcela significativa dos acidentes. Entretanto, a frequência de uma causa não indica necessariamente que ela esteja associada a maior gravidade. Por esse motivo, as principais causas serão posteriormente analisadas em relação à ocorrência de acidentes fatais.

In [ ]:
# 8.5.7 — Principais causas e acidentes fatais

top_causas_10 = df['causa_acidente'].value_counts().head(10).index

causas_gravidade = (
    df[df['causa_acidente'].isin(top_causas_10)]
    .groupby('causa_acidente')
    .agg(
        acidentes=('id', 'count'),
        mortos=('mortos', 'sum'),
        acidentes_fatais=('mortos', lambda x: (x > 0).sum())
    ).reset_index()
)

causas_gravidade['percentual_fatais'] = (
    causas_gravidade['acidentes_fatais'] / causas_gravidade['acidentes'] * 100)

causas_gravidade = causas_gravidade.sort_values(
    'percentual_fatais',
    ascending=False
)

print(causas_gravidade)

                                      causa_acidente  acidentes  mortos  \
8                             Transitar na contramão       7300    2743   
9                            Velocidade Incompatível      12675    1382   
2                                  Condutor Dormindo       6346     590   
1                     Ausência de reação do condutor      31459    2288   
0  Acessar a via sem observar a presença dos outr...      20380    1205   
7           Reação tardia ou ineficiente do condutor      31574    1838   
6                        Manobra de mudança de faixa      12142     688   
5                   Ingestão de álcool pelo condutor      11138     593   
3  Condutor deixou de manter distância do veículo...      13145     284   
4               Demais falhas mecânicas ou elétricas       9715     198   

   acidentes_fatais  percentual_fatais  
8              2114          28.958904  
9              1135           8.954635  
2               458           7.217145  
1         

#### 8.5.7.1 — Resultado da análise das principais causas e gravidade

A análise das dez causas de acidentes mais frequentes permitiu comparar o volume de ocorrências com indicadores de gravidade.

A causa **"Reação tardia ou ineficiente do condutor"** apresentou o maior número de acidentes, com 31.574 registros, seguida por **"Ausência de reação do condutor"**, com 31.459. Apesar da elevada frequência, essas causas apresentaram proporções de acidentes fatais de 5,09% e 6,57%, respectivamente.

Em contraste, **"Transitar na contramão"** apresentou 7.300 acidentes, mas registrou a maior proporção de acidentes fatais entre as dez principais causas, com **28,96%**. Também se destacaram **"Velocidade Incompatível"**, com 8,95%, e **"Condutor Dormindo"**, com 7,22%.

Os resultados demonstram que as causas mais frequentes não são necessariamente aquelas associadas às maiores proporções de acidentes fatais. Dessa forma, a análise conjunta de frequência e gravidade permite uma compreensão mais adequada dos padrões encontrados na base.

Os resultados representam associações observadas nos dados e não permitem afirmar, isoladamente, uma relação de causalidade entre determinada causa e a ocorrência de mortes.

#### 9. Principais Insights

A análise exploratória dos acidentes registrados em rodovias federais brasileiras entre 2023 e 2025 permitiu identificar padrões relacionados ao volume de ocorrências, gravidade, distribuição geográfica e causas dos acidentes.

Os principais insights identificados foram:

#### 9.1 — Volume e evolução dos acidentes

Foram registrados 213.451 acidentes entre 2023 e 2025. O maior volume ocorreu em 2024, com 73.156 registros, seguido por uma pequena redução em 2025, que apresentou 72.529 ocorrências.

Apesar dessa variação, a proporção de acidentes fatais permaneceu estável nos três anos, próxima de 7,1%, indicando que não houve uma mudança relevante nesse indicador durante o período analisado.

#### 9.2 — Gravidade dos acidentes

Dos 213.451 acidentes analisados, 15.290 foram fatais, correspondendo a 7,16% das ocorrências.

No período, foram registradas 17.830 mortes e 246.539 feridos. Dessa forma, além do volume de acidentes, a gravidade das ocorrências representa uma dimensão importante para compreender o problema.

#### 9.3 — Diferenças entre os estados

Minas Gerais apresentou o maior número absoluto de acidentes, com 27.873 ocorrências.

Entretanto, quando considerada a proporção de acidentes fatais, o cenário foi diferente. Maranhão apresentou a maior proporção, com 19,59%, seguido por Pará, com 18,92%. A Bahia também apresentou uma proporção elevada, de 11,61%.

Esse resultado demonstra que o estado com maior quantidade de acidentes não necessariamente apresenta a maior gravidade proporcional.

#### 9.4 — Frequência das causas x gravidade

As causas mais frequentes foram "Reação tardia ou ineficiente do condutor", com 31.574 ocorrências, e "Ausência de reação do condutor", com 31.459.

Por outro lado, entre as dez causas mais frequentes, "Transitar na contramão" apresentou a maior proporção de acidentes fatais, com 28,96%, seguida por "Velocidade Incompatível", com 8,95%, e "Condutor Dormindo", com 7,22%.

Esse resultado demonstra que as causas mais frequentes não são necessariamente aquelas associadas aos acidentes de maior gravidade.

#### 9.5 — Síntese

Os principais achados da análise foram:

- 213.451 acidentes registrados entre 2023 e 2025;
- 15.290 acidentes fatais, correspondendo a 7,16% do total;
- 17.830 mortes e 246.539 feridos;
- 2024 apresentou o maior volume de acidentes;
- Minas Gerais apresentou o maior número absoluto de ocorrências;
- Maranhão e Pará apresentaram as maiores proporções de acidentes fatais;
- causas relacionadas ao comportamento e à reação dos condutores aparecem entre as mais frequentes;
- "Transitar na contramão" apresentou destaque pela elevada proporção de acidentes fatais;
- frequência e gravidade apresentaram comportamentos diferentes, demonstrando a importância de analisar as duas dimensões conjuntamente.

Esses resultados servirão como base para a construção do dashboard no Looker Studio, permitindo explorar os principais padrões encontrados de forma visual e interativa.

#### 10. Sugestões de ações

Os resultados encontrados indicam que estratégias de prevenção podem considerar não apenas o volume de acidentes, mas também a gravidade das ocorrências e suas principais características.

Entre as possíveis ações estão:

Direcionamento de ações educativas e de fiscalização para comportamentos associados a acidentes de maior gravidade, especialmente aqueles relacionados à condução na contramão e à velocidade incompatível.
Análise regionalizada das ocorrências, considerando estados com maior proporção de acidentes fatais para identificar locais que possam demandar maior atenção preventiva.
Monitoramento contínuo dos indicadores de acidentes fatais, permitindo acompanhar mudanças na gravidade das ocorrências ao longo do tempo.
Utilização dos dados para priorização de ações preventivas, combinando volume de acidentes, proporção de fatalidade e características das ocorrências.

Essas ações devem ser interpretadas como possibilidades baseadas nos padrões observados na análise. Os dados utilizados permitem identificar associações e padrões, mas não são suficientes, isoladamente, para determinar relações de causalidade.

In [ ]:
df.to_csv(
    '/content/drive/MyDrive/Colab Notebooks/projeto semantix/acidentes_2023_2025.csv',
    index=False,
    encoding='utf-8-sig'
)

print('Arquivo CSV criado com sucesso!')

Arquivo CSV criado com sucesso!
